sktime/annotation/hmm_learn.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+"""
+Hidden Markov Model based annotation from hmmlearn.
+
+This code provides a base interface template for models
+from hmmlearn for using that library for annotation of time series.
+
+Please see the original library
+(https://github.com/hmmlearn/hmmlearn/blob/main/lib/hmmlearn/hmm.py)
+"""
+
+import pandas as pd
+from attrs import define
+
+from sktime.annotation.base import BaseSeriesAnnotator
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+__author__ = ["miraep8"]
+__all__ = ["BaseHMMLearn", "GaussianHMM"]
+
+
+class BaseHMMLearn(BaseSeriesAnnotator):
+    """Base class for all HMM wrappers, handles required overlap between packages."""
+
+    _tags = {"univariate-only": True, "fit_is_empty": True}  # for unit test cases
+    _hmm_estimator = None
+
+    def __init__(self):
+        super(BaseHMMLearn, self).__init__()
+
+    def _fit(self, X, Y=None):
+        series = isinstance(X, pd.Series)
+        if series:
+            X = (X.to_numpy()).reshape((-1, 1))
+        self._hmm_estimator = self._hmm_estimator.fit(X)
+        return self
+
+    def _predict(self, X):
+        series = isinstance(X, pd.Series)
+        if series:
+            index = X.index
+            X = (X.to_numpy()).reshape((-1, 1))
+        X_prime = self._hmm_estimator.predict(X)
+        if series:
+            X_prime = pd.Series(X_prime, index=index)
+        return X_prime
+
+    def sample(self, n_samples=1, random_state=None, currstate=None):
+        """Interface class which allows users to sample from their HMM."""
+        return self._hmm_estimator.sample(n_samples, random_state, currstate)
+
+
+_check_soft_dependencies("hmmlearn.hmm", severity="warning")
+
+
+@define
```

Comment:
hm, you *had* to sneak this in, didn't you?
I suppose it's ok in an experimental part of sktime, let's see whether this causes any problems...

Can you check whether the top level of the test suite is executed on `GaussianHMM`? This should be the case if it ultimately inherits from `BaseObject`.

Reply:
Yes, I do believe it ultimately ends up running the test suite on `GaussianHMM` given the fact that I had to debug a few of them :)